equinix/data_source_metal_project_acc_test.go

```diff
@@ -38,6 +38,12 @@ func TestAccDataSourceMetalProject_basic(t *testing.T) {
 
 func testAccDataSourceMetalProject_basic(r string) string {
 	return fmt.Sprintf(`
+terraform {
```

Comment:
I didn't see evidence of the module_name making it to the Provider Configure in this PR.
```
export TF_PROVIDER_LOG=debug; export PACKNGO_DEBUG=1; 
make testacc TESTS_REGEXP='(TestAccDataSourceMetalProject_basic)'
```

Reply:
After some tests I confirm that:

- GetProviderMeta returns nil during provider configuration, so a build user-agent function must be included in all create/read/update/delete resource functions
- Only works with resources and not with datasources
- Works fine with both tf files and modules